### ISSUE TITLE
`WRONG_OVERLOADING_FUNCTION_ARGUMENTS` triggers with visibility differences

### ISSUE THREAD

**Issue by jgberry** [saveourtool/diktat#1505]
## Describe the bug
`WRONG_OVERLOADING_FUNCTION_ARGUMENTS` is triggered even if the functions have differing visibility and/or modifiers (abstract, suspend, etc.)

Example with different visibility:
```kotlin
public fun doStuff() {
    doStuff(1)
}

private fun doStuff(x: Int) {
    println(x)
}
```

Example with inline:
```kotlin
public fun doStuff() {
    doStuff(1)
}

public inline fun doStuff(x: Int) {
    println(x)
}
```

## Expected behavior
`WRONG_OVERLOADING_FUNCTION_ARGUMENTS` should only trigger if the function have identical visibility and identical modifiers.

## Observed behavior
`WRONG_OVERLOADING_FUNCTION_ARGUMENTS` ignores function visibility and modifiers.

## Steps to Reproduce
See above examples.

## Environment information
* diktat version: 1.2.3
* build tool (maven/gradle): gradle
* how is diktat run (CLI, plugin, etc.): ./gradlew diktatCheck
* kotlin version: 1.7.10
* operating system: Ubuntu
* link to a project (if your project is public):

**Comment by akuleshov7** [comment]
Hello! thank you for your report, yes, definitely a bug - should be fixed

**Comment by akuleshov7** [comment]
I am even surprise that we are so smart that we have such inspection 😄

**Comment by akuleshov7** [comment]
@jgberry thank you for your ideas, we fixed it and will take it to the next release
